bin/distmap

```diff
@@ -20,124 +20,166 @@ use DownloadTrimmedRead;
+# ARGUMENT VARIABLES
+## REQUIRED
 my $reference_fasta="";
-my $refindex_archive="";
-my @input_files = ();
 my $output_directory="";
+my @input_files = ();
+my $mapper_args="";
+my @mapper=();
+my @mapper_path=();
+my @mapper_args=();
+
+## PSEUDO-REQUIRED
+### TODO: should this be called after parsing if not provided? I think that it would be more consistent.
+my $hadoop_home=$ENV{"HADOOP_HOME"};
+my $picard=`which picard`;
+my $readtools_arg = `which readtools`;
 
+## OPTIONAL
+### pipeline
 my $only_index=0;
 my $only_process=0;
-my $only_trim=0;
-my $no_trim=0;
-my $only_map=0;
 my $only_hdfs_upload=0;
+my $only_map=0;
 my $only_hdfs_download =0;
 my $only_merging=0;
 my $only_delete_temp=0;
-my $only_download_reads=0;
-
-my $gsnap_output_split=0;
-#my $hadoop_home=$ENV{"HADOOP_HOME"};
-my $hadoop_home=$ENV{"HADOOP_HOME"};
-
-my $mapper="";
-my $mapper_path="";
-my $mapper_args="";
-my @mapper=();
-my @mapper_path=();
-my @mapper_args=();
-
-my $bwa_sampe_args="";
-my $picard=`which picard`;
-my $picard_mergesamfiles_jar=""; # OBSOLETE
-my $picard_sortsam_jar=""; # OBSOLETE
-my $picard_mark_duplicates_jar="";
-my $readtools_arg = `which readtools`; # TODO: should this be called after parsing if not provided? I think that it is more consistent.
+my $trim_args="";
+### others
 my $tmp_dir = "";
 my $output_format="bam";
-my $job_desc="";
+my $refindex_archive="";
 my $queue_name="pg1";
-my $job_priority="VERY_HIGH";
-my $hadoop_scheduler="Fair";
-
-my $mappers_exe_archive = "";
-
-my $trim_script_path="";
-my $trim_args=""; # TODO: standard readtools trimming args
+my $job_desc="";
+my $gsnap_output_split=0; # TODO - make conditional
+my $bwa_sampe_args=""; # TODO - make conditional
 
-my $verbose = 0;
-my $test = 0;
 my $help = 0;
+my $version = 0;
 
+## DEPRECATED (log warning if used?) - TODO: remove
+my $no_trim=0;
+my $qualThreshold=20;
+my $discardRemainingNs=0;
+my $minLength=40;
+my $trimQuality=0; ## TODO: be careful with this argument (it is really no-trim-quality switch)
+my $no5ptrim=0;
+my $verbose = 0;
 
+## OBSOLETE (log warning if used?) - TODO: remove
+my $picard_mergesamfiles_jar="";
+my $picard_sortsam_jar="";
+my $picard_mark_duplicates_jar=""; # TODO: is not even used - remove
+my $only_trim=0;
+my $trim_script_path="";
+my $fastqtype="illumina";
+my $nozip=0;
 
-#### Trimming parameters
+## UNDOCUMENTED - TODO: documment
+my $only_download_reads=0;
+my $mappers_exe_archive = "";
 
-    my $qualThreshold=20;
-    my $minLength=40;
-    my $fastqtype="illumina";
-    my $discardRemainingNs=0;
-    my $trimQuality=0; ## TODO: be careful with this argument (it is really no-trim-quality switch)
-    my $no5ptrim=0;
-    my $nozip=0;
+## TODO - supposedly options but not in the GetOptions block!!! cannot be specified
+## TODO - add to the parsing or remove support for them and default always!
+## TODO - they appear in the example!!
+my $job_priority="VERY_HIGH";
+my $hadoop_scheduler="Fair"; # this is not used at all (remove?)
```

Comment:
I believe there are scheduler-dependent -D... options that could be passed. I do not think it is currently used, and also it is plain wrong to set it here (would need to query the Hadoop config for the scheduler type).

Reply:
This was already in, and I just kept it like that. Should I just remove?

Reply:
The `$hadoop_scheduler` is completely unused and I removed it in the next commit (and also from the help)

For the `$job_priority`, I open issue #82